### PR TITLE
csv-diff: add page

### LINKED ISSUE #8444: page request: csv-diff
### Command description

https://github.com/simonw/csv-diff

### Command details

_No response_

### OS

- [ ] Android
- [X] Linux
- [X] macOS
- [ ] sunOS
- [X] Windows

### PR DESCRIPTION
**Pull request by gutjuri**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):**

closes #8444

### CODE REVIEW

File: pages/common/csv-diff.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# csv-diff
+
+> View differences between two CSV, TSV or JSON files.
+> More information: <https://github.com/simonw/csv-diff>.
+
+- Show a human-readable summary of differences between files, treating the column `key` as an unique identifier:
```
Comment (navarroaxel) [inline pages/common/csv-diff.md:6]: Perhaps this better
```suggestion
- Show a human-readable summary of differences between files using a specific column as an unique identifier:
```

File: pages/common/csv-diff.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{key}}`
```
Comment (navarroaxel) [inline pages/common/csv-diff.md:8]:
```suggestion
`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}}`
```

File: pages/common/csv-diff.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+- Show a human-readable summary of differences between files that includes unchanges values in rows that have at least one change:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{key}} --show-unchanged`
+
+- Show a machine-readable summary of differences between files, treating the column `key` as an unique identifier:
```
Comment (navarroaxel) [inline pages/common/csv-diff.md:14]:
```suggestion
- Show a summary of differences between files in JSON format using a specific column as an unique identifier:
```

File: pages/common/csv-diff.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+> View differences between two CSV, TSV or JSON files.
+> More information: <https://github.com/simonw/csv-diff>.
+
+- Show a human-readable summary of differences between files, treating the column `key` as an unique identifier:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{key}}`
+
+- Show a human-readable summary of differences between files that includes unchanges values in rows that have at least one change:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{key}} --show-unchanged`
```
Comment (navarroaxel) [inline pages/common/csv-diff.md:12]:
```suggestion
`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}} --show-unchanged`
```

File: pages/common/csv-diff.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}}`
+
+- Show a human-readable summary of differences between files that includes unchanges values in rows that have at least one change:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}} --show-unchanged`
+
+- Show a summary of differences between files in JSON format using a specific column as an unique identifier:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{key}} --json`
```
Comment (navarroaxel) [inline pages/common/csv-diff.md:16]:
```suggestion
`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}} --json`
```

**Review by navarroaxel — COMMENTED** [review]
Remember that you can apply suggestions in [batch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request).

**Comment by gutjuri** [comment]
> Remember that you can apply suggestions in [batch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request).

Thanks, I wasn't aware.

**Review by navarroaxel — APPROVED** [review]

File: pages/common/csv-diff.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+- Show a human-readable summary of differences between files using a specific column as an unique identifier:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}}`
+
+- Show a human-readable summary of differences between files that includes unchanges values in rows that have at least one change:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}} --show-unchanged`
+
+- Show a summary of differences between files in JSON format using a specific column as an unique identifier:
```
Comment (kbdharun) [inline pages/common/csv-diff.md:14]:
```suggestion
- Display a summary of differences between files in JSON format using a specific column as a unique identifier:
```

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution, I have some suggestions for command descriptions. 

@navarroaxel should we use `path/to/file1.csv`, etc here or is the current one fine?

**Comment by navarroaxel** [comment]
> Thanks for your contribution, I have some suggestions for command descriptions.
> 
> @navarroaxel should we use `path/to/file1.csv`, etc here or is the current one fine?

@kbdharun I don't have an strong opinion. `delta` and `diff` use `old_file` and `new_file` but for this command `file1` and `file2` are better. Or `path/to/file1.csv` and `path/to/file2.csv`

**Review by kbdharun — APPROVED** [review]

File: pages/common/csv-diff.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# csv-diff
+
+> View differences between two CSV, TSV or JSON files.
+> More information: <https://github.com/simonw/csv-diff>.
+
+- Display a human-readable summary of differences between files using a specific column as a unique identifier:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}}`
```
Comment (navarroaxel) [inline pages/common/csv-diff.md:8]:
```suggestion
`csv-diff {{path/to/file1.csv}} {{path/to/file2.csv}} --key={{column_name}}`
```

File: pages/common/csv-diff.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+- Display a human-readable summary of differences between files that includes unchanged values in rows with at least one change:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}} --show-unchanged`
```
Comment (navarroaxel) [inline pages/common/csv-diff.md:12]:
```suggestion
`csv-diff {{path/to/file1.csv}} {{path/to/file2.csv}} --key={{column_name}} --show-unchanged`
```

File: pages/common/csv-diff.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+- Display a summary of differences between files in JSON format using a specific column as a unique identifier:
+
+`csv-diff {{one.csv}} {{two.csv}} --key={{column_name}} --json`
```
Comment (navarroaxel) [inline pages/common/csv-diff.md:16]:
```suggestion
`csv-diff {{path/to/file1.csv}} {{path/to/file2.csv}} --key={{column_name}} --json`
```

**Review by mfrw — APPROVED** [review]
Thank you @gutjuri 
LGTM :)

**Pull request merged**